### ISSUE TITLE
Cleanup All Code crashes VS2015 on large solution (3310 files)

### ISSUE THREAD

**Issue by jakubsuchybio** [codecadwallader/codemaid#209]
Hi,
I was using CodeMaid for 310 files big c# project (client) and it went ok. Today I ran it on 3310 files big c# project (server) and it crashed my VS2015 3 times already. Always between 100-150 files went fine and then crashed.
Now I changed a config a bit with Diagnostics mode ON. 
(specifics see here: https://gist.github.com/jakubsuchybio/a4a7f0f2e29bc570ed93)
Ran Cleanup All Code again, it crashed like this:
![crash4](https://cloud.githubusercontent.com/assets/6567287/12550145/15d1cdca-c362-11e5-940a-bb0abd8ace20.png)
Also last 2 crashes were on the same file Perks.cs, but when I tried to run Cleanup only on Perks.cs, it works fine.
And I can't find any diagnostics log.
From here I guess I need your help guys. To debug it more. :)

**Comment by codecadwallader** [comment]
Thanks for reporting the issue.  There is a known issue that Visual Studio's memory grows with every file you open.  You can see this even with CodeMaid not installed, the more files you open, the more Visual Studio's memory footprint increases.

We've repeatedly confirmed that CodeMaid does not have any memory leaks during large cleanups, but can confirm there are un-reclaimed objects within Visual Studio itself every time a document is opened.  I can't say if that's deliberate caching, or just standard memory leaks.  More details on #83 .

When you run a large cleanup, you can end up opening hundreds or thousands of files and as your screenshot shows memory starts to get out of control.  Visual Studio is a 32-bit application so it does have an upper bound around 2GB before it buckles.

Unfortunately at this time we don't have a good workaround, other than to clean up parts of the solution separately and bounce Visual Studio in-between.  I would love to have a better approach and have marked this card as "help wanted" in case anybody has some better ideas.

**Comment by jakubsuchybio** [comment]
Sory for OT: I was cleaning up our codebase again with CodeMaid and I was doing smaller chunks, but wanted to speed up a process. So... Just for fun:
![image](https://cloud.githubusercontent.com/assets/6567287/13284636/d522f5d2-daf5-11e5-9a62-a636c9ae4c89.png)
This is what I call Manual Paralelization :)

**Comment by codecadwallader** [comment]
Haha, that's pretty cool.  Thanks for sharing. :)

**Comment by TiagoGoddard** [comment]
> Unfortunately at this time we don't have a good workaround, other than to clean up parts of the solution separately and bounce Visual Studio in-between. I would love to have a better approach and have marked this card as "help wanted" in case anybody has some better ideas.

We could have some kind of "continue from last crash" where we inform the number of the file we last stopped, so even if it crashes, we can skip to the last file it knew it was cleaning?

**Comment by codecadwallader** [comment]
> We could have some kind of "continue from last crash" where we inform the number of the file we last stopped, so even if it crashes, we can skip to the last file it knew it was cleaning?
Yeah that's a good idea.. some kind of operational log that starts with batch operations.  If Visual Studio starts and the operational log is incomplete we could notify users and they could choose to resume or cancel.
